pkg/agent/openflow/packetin.go

```diff
@@ -68,28 +93,28 @@ const (
 	PacketInQueueRate = 100
 )
 
-// RegisterPacketInHandler stores controller handler in a map of map with reason and name as keys.
-func (c *client) RegisterPacketInHandler(packetHandlerReason uint8, packetHandlerName string, packetInHandler interface{}) {
+// RegisterPacketInHandler stores controller handler in a map of map with category and name as keys.
+func (c *client) RegisterPacketInHandler(packetHandlerCategory uint8, packetHandlerName string, packetInHandler interface{}) {
 	handler, ok := packetInHandler.(PacketInHandler)
 	if !ok {
 		klog.Errorf("Invalid controller to handle packetin.")
 		return
 	}
-	if c.packetInHandlers[packetHandlerReason] == nil {
-		c.packetInHandlers[packetHandlerReason] = map[string]PacketInHandler{}
+	if c.packetInHandlers[packetHandlerCategory] == nil {
```

Comment:
Do we still need this struct `map[uint8]map[string]PacketInHandler` for pakcetInHandlers? I mean after using userData as the filter, do we still need to support multiple handers per packetIn category?

Reply:
Good one. Changed to `map[uint8]PacketInHandler`.